File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsMastery.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class MasteryConcept(
    val id: String,
    val name: String,
)

@Serializable
data class MasteryCell(
    val conceptId: String,
    val masteryScore: Double? = null,
    val assessed: Boolean = false,
    val updatedAt: String? = null,
)

@Serializable
data class StudentMasteryRow(
    val enrollmentId: String,
    val userId: String,
    val concepts: List<MasteryConcept> = emptyList(),
    val cells: List<MasteryCell> = emptyList(),
)

@Serializable
data class ReportCardSummary(
    val id: String,
    val studentId: String,
    val courseId: String,
    val gradingPeriod: String,
    val status: String,
    val finalGradePct: Double? = null,
    val letterGrade: String? = null,
    val comment: String? = null,
    val pdfUrl: String? = null,
    val generatedAt: String? = null,
    val releasedAt: String? = null,
    val createdAt: String? = null,
)

@Serializable
data class MyReportCardsResponse(
    val reportCards: List<ReportCardSummary> = emptyList(),
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsMastery.swift
import Foundation

struct MasteryConcept: Codable, Identifiable, Hashable {
    var id: String
    var name: String
}

struct MasteryCell: Codable, Hashable {
    var conceptId: String
    var masteryScore: Double?
    var assessed: Bool
    var updatedAt: String?
}

struct StudentMasteryRow: Codable {
    var enrollmentId: String
    var userId: String
    var concepts: [MasteryConcept]
    var cells: [MasteryCell]
}

struct ReportCardSummary: Codable, Identifiable, Hashable {
    var id: String
    var studentId: String
    var courseId: String
    var gradingPeriod: String
    var status: String
    var finalGradePct: Double?
    var letterGrade: String?
    var comment: String?
    var pdfUrl: String?
    var generatedAt: String?
    var releasedAt: String?
    var createdAt: String?
}

struct MyReportCardsResponse: Decodable {
    var reportCards: [ReportCardSummary]
}
